From 8b275b3e98acb8c1d600c4d8c6b78953613a58ac Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Tue, 3 Jul 2018 10:54:03 +0800
Subject: [PATCH 11/30] qt5base: fix QT eglfs display bug

The app occur Segmentation fault when it exit and click screen

Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 src/gui/kernel/qscreen.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui/kernel/qscreen.cpp b/src/gui/kernel/qscreen.cpp
index ad128a88..166c1297 100644
--- a/src/gui/kernel/qscreen.cpp
+++ b/src/gui/kernel/qscreen.cpp
@@ -385,6 +385,9 @@ QSize QScreen::availableSize() const
 */
 QRect QScreen::geometry() const
 {
+    if (!qApp)
+       return QRect(0, 0, 0, 0);
+
     Q_D(const QScreen);
     return d->geometry;
 }
-- 
2.20.1

